src/blocks/pomodoro.rs

```diff
@@ -0,0 +1,338 @@
+    pub fn on_start(&mut self, _: Start) -> Started {
+        Started {
+            seconds: self.seconds,
+            text: "\u{f04b}".to_string(),
+        }
+    }
+
+    pub fn on_stop(&mut self, _: Stop) -> Stopped {
+        Stopped { text: "\u{25a0} 0:00".to_string() }
+    }
+}
+
+impl Breaking {
+    pub fn on_stop(&mut self, _: Stop) -> Stopped {
+        Stopped { text: "\u{25a0} 0:00".to_string() }
+    }
+
+    pub fn on_start(&mut self, _: Start) -> Started {
+        Started {
+            seconds: 0,
+            text: "\u{f04b}".to_string(),
+        }
+    }
+}
+
+methods!(State,
+  [
+    Stopped, Started, Paused, Breaking => fn get_text(&self) -> String,
+    Started, Breaking => get seconds: usize
+  ]
+);
+
+impl Stopped {
+    pub fn get_text(&self) -> String {
+        self.text.to_owned()
+    }
+}
+
+impl State {
+    pub fn tick(&mut self) -> Option<()> {
+        match self {
+            State::Started(ref mut v) => Some(v.tick()),
+            State::Breaking(ref mut v) => Some(v.tick()),
+            _ => None,
+        }
+    }
+}
+
+impl Started {
+    pub fn get_text(&self) -> String {
+        format!("{} {}:{:02}", self.text, self.seconds / 60, self.seconds % 60)
+    }
+
+    pub fn tick(&mut self) -> () {
+        self.seconds = self.seconds + 1;
+    }
+}
+
+impl Paused {
+    pub fn get_text(&self) -> String {
+        format!("{} {}:{:02}", self.text, self.seconds / 60, self.seconds % 60)
+    }
+}
+
+impl Breaking {
+    pub fn get_text(&self) -> String {
+        format!("{} {}:{:02}", self.text, self.seconds / 60, self.seconds % 60)
+    }
+
+    pub fn tick(&mut self) -> () {
+        self.seconds = self.seconds + 1;
+    }
+}
+
+pub struct Pomodoro {
+    id: String,
+    time: TextWidget,
+    state: State,
+    length: usize,
+    break_length: usize,
+    update_interval: Duration,
+    count: usize,
+}
+
+#[derive(Deserialize, Debug, Default, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct PomodoroConfig {
+    #[serde(default = "PomodoroConfig::default_length")]
+    pub length: usize,
+    #[serde(default = "PomodoroConfig::default_break_length")]
+    pub break_length: usize,
+}
+
+impl PomodoroConfig {
+    fn default_length() -> usize {
+        25
+    }
+
+    fn default_break_length() -> usize {
+        5
+    }
+}
+
+impl ConfigBlock for Pomodoro {
+    type Config = PomodoroConfig;
+
+    fn new(block_config: Self::Config, config: Config, _send: Sender<Task>) -> Result<Self> {
+        let id: String = Uuid::new_v4().simple().to_string();
+        let id_copy = id.clone();
+
+        Ok(Pomodoro {
+            id: id_copy,
+            time: TextWidget::new(config).with_icon("pomodoro"),
+            state: State::stopped("\u{25a0} 0:00".to_string()),
+            length: block_config.length * 60,             // convert to minutes
+            break_length: block_config.break_length * 60, // convert to minutes
+            update_interval: Duration::from_millis(1000),
```

Comment:
a timer that lasts 25 minutes that doesn't update every second didn't seem a requirement. but if you think it is I can do the approach you suggested and compare times instead. it will just be a bunch more code I think